query-syntax/VirgoQueryLexer.g4

```diff
@@ -0,0 +1,29 @@
+// Define a grammar called Query
+lexer grammar VirgoQueryLexer;
+fragment WS : [ \t\r\n]+;
+LPAREN : '(' ;
+RPAREN : ')' ;
+
+BOOLEAN : ('AND'|'OR'|'NOT') ;
+COLON : ':' ;
+TITLE : 'title';
+AUTHOR : 'author';
+SUBJECT : 'subject';
+KEYWORD : 'keyword';
+LBRACE : '{' -> pushMode(SEARCH);
+WS1: WS -> skip;
+
+mode SEARCH;
+
+QUOTE : '"' ;
+LPAREN2 : '(' ->type(LPAREN);
+RPAREN2 : ')' ->type(RPAREN);
+LBRACKET : '[' ;
+RBRACKET : ']' ;
+COMMA : ',' ;
+RBRACE : '}' -> popMode;
+BOOLEAN1 : ('AND'|'OR'|'NOT') ->type(BOOLEAN) ;
+SEARCH_WORD :  [\p{L}\p{Nl}\p{Nd}]+ ;
```

Comment:
If I'm reading this correctly, we allow any letter (including spaces, asterisks, accents, etc) and decimal digits and letter-like numeric characters.    I'm not sure exactly what we're excluding, but tweaking the allowable characters shouldn't require significant changes (just a library update) to the application.  

The code that transforms these queries into Solr will have to follow it's escaping rules and enforce (or report) it's disallowed character rules.

Reply:
I think that this:    [\p{L}\p{Nl}\p{Nd}]+    matches Unicode Letters and decimal digits and "letter-like numeric characters"  but does not include spaces or any whitespace.   It also does not match punctuation characters, and may not match combining accent characters, (Although it probably should, since it does match Unicode letters that include accents )

You are correct that tweaking the set of characters accepted by as a SEARCH_WORD would be easy to tweak, and most of those sort of changes would not require significant changes.  Most would only require re-generating the parser code.